### DISCUSSION TITLE
Help: How can i get a newsimage url in the news_grid_template.php?

### DISCUSSION THREAD

**Discussion by Alex-e107nl** [e107inc/e107#5149] [Q&A Core] [answered]
When i want to use the first news image in the news_grid_template.php i can use {NEWS_IMAGE: type=tag&class=media-object img-rounded&placeholder=1} or just {NEWS_IMAGE} etc but what do i use when i only want the url in case you need it for style="background-image: url('unsplash-photo-1.jpg');" ? Believe me, i have tried a lot :-) 

Is there an overview of the options somewhere?

Help please?

**Comment by Jimmi08** [comment, marked as answer]
**NOT TESTED!** 

EDITED left only working solution 

Legacy shortcode (it should still work)
`{NEWSIMAGE: item=1&type=src}`

It was replaced with  the new image shortcode, but there is no item parameter
`{NEWS_IMAGE: type=src&placeholder=true}`

**Comment by Alex-e107nl** [comment]
Thx @Jimmi08, i have tested it but none of the options just returns a simple url...

Legacy shortcode (it should still work)
`{NEWSIMAGE: item=1&type=src}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" demo08="" theme="" img="" 0x0="" bootstrap5="" images="" superhero.png");"="">`

It was replaced with the new image shortcode, but there is no item parameter
`{NEWS_IMAGE: type=src&placeholder=true}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" demo08="" theme="" img="" 0x0="" bootstrap5="" images="" superhero.png");"="">`
													 

To get the first image use the shortcode:
`{NEWS_THUMBNAIL}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" <a="" href="/demo08/news/item-3.html"><img class="news_image img-responsive img-fluid img-rounded rounded news-thumbnail-1" src="/demo08/thumb.php?src=%2Fdemo08%2Fe107_themes%2Fbootstrap5%2Fimages%2Fsuperhero.png&amp;aw=0&amp;ah=0" alt="" style="">");"&gt;`
													

It should return the path with:
`{NEWS_THUMBNAIL=src=}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" <a="" href="/demo08/news/item-3.html"><img class="news_image img-responsive img-fluid img-rounded rounded news-thumbnail-1" src="/demo08/thumb.php?src=%2Fdemo08%2Fe107_themes%2Fbootstrap5%2Fimages%2Fsuperhero.png&amp;aw=0&amp;ah=0" alt="" style="">");"&gt;`

if it returns a too small image, try to set full size with (not clear what is correct):

`{NEWS_THUMBNAIL=src|w=0&h=0}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" demo08="" thumb.php?src="%2Fdemo08%2Fe107_themes%2Fbootstrap5%2Fimages%2Fsuperhero.png&amp;w=0&amp;h=0" ");"="">`
	 												
														
`{NEWS_THUMBNAIL=src|aw=0&ah=0}`

Gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" demo08="" thumb.php?src="%2Fdemo08%2Fe107_themes%2Fbootstrap5%2Fimages%2Fsuperhero.png&amp;aw=0&amp;ah=0&quot;);&quot;">`


It should be possible to just get the url to the image i think? For the {CMENUIMAGE} you can at least do {CMENUIMAGE=url} 

I am no coder, just want to build a nice theme :-)

**Reply by Jimmi08** [reply]
src for image it should be the same as URL parameter for cmenuimage.  I use it all time... weird. 

```
		switch($parms[1])
		{
			case 'src':
				return $src;
			break;

			case 'tag':
				return "<img class='news_image ".$class."' src='".$src."' alt='' style='".$style."' {$dimensions} {$srcset} />";
			break;

			case 'img':
				return "<a href='".$_src."' rel='external image'><img class='news_image ".$class."' src='".$src."' alt='' style='".$style."' {$dimensions} {$srcset} /></a>";
			break;

			default:
				return "<a href='".e107::getUrl()->create('news/view/item', $this->news_item)."'><img class='news_image img-responsive img-fluid img-rounded rounded ".$class."' src='".$src."' alt='' style='".$style."' {$dimensions} {$srcset} /></a>";
			break;
		}
```

For sure, the src parameter should return the path only to the image.  Maybe this is documentation wrong, try this last time (I can't now)
`{NEWS_THUMBNAIL: src=1&w=0&h=0}`

If it is not working, then post the template - you know where - and I will do some tests myself.   Or look at some of my themes.
It has to work! This is basic.

**Reply by Alex-e107nl** [reply]
I have posted the template and will look further in your themes... thx!

**Comment by Alex-e107nl** [comment]
So when I do the next thing in news_grid_template.php:

```
$nimage = "/demo08/media/img/0x0/2023-12/02_dl_h1080.jpg";
style="background-image: url("'.$nimage.'");">
```

It gives:

`<div class="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg" style="background-image: url(" demo08="" media="" img="" 0x0="" 2023-12="" 02_dl_h1080.jpg");"="">`

**Reply by Jimmi08** [reply]
I am on it now. Sec.

**Reply by Jimmi08** [reply]
Fixed. problem was somewhere else, not in the shortcode itself.

**Reply by Alex-e107nl** [reply]
Has anyone ever said that you are really great! You are awesome! It functions!

**Comment by Alex-e107nl** [comment]
Oh almost forgot, the correct answer to my question was given by @Jimmi08 and it is.....

`style="background-image: url({NEWS_IMAGE: type=src});`

I also hate reading a discussion and not seeing the right solution... :-)
